Core/include/Acts/Vertexing/AdaptiveMultiVertexFinder.hpp

```diff
@@ -312,21 +341,27 @@ class AdaptiveMultiVertexFinder {
   /// @param vtx The vertex candidate
   /// @param[out] seedTracks The seed tracks
   /// @param fitterState The vertex fitter state
+  /// @param[out] removedSeedTracks Collection of seed track that will be
+  /// removed
   void removeCompatibleTracksFromSeedTracks(
       Vertex<InputTrack_t>& vtx, std::vector<const InputTrack_t*>& seedTracks,
-      FitterState_t& fitterState) const;
+      FitterState_t& fitterState,
+      std::vector<const InputTrack_t*>& removedSeedTracks) const;
 
   /// @brief Method that tries to remove a non-compatible track
   /// from seed tracks after removing a compatible track failed.
   ///
   /// @param vtx The vertex candidate
   /// @param[out] seedTracks The seed tracks
   /// @param fitterState The vertex fitter state
+  /// @param[out] removedSeedTracks Collection of seed track that will be
+  /// removed
   ///
   /// @return Non-compatible track was removed
   bool canRemoveNonCompatibleTrackFromSeedTracks(
```

Comment:
I think the naming is misleading, as it return a `bool` and the method is called `Toddler::canHaveAnApple(...)` I would think the method only checks if I can have an apple or not - I would not expect the `toddler` to eat the apple within this call.

Reply:
Yes, you're right. When I introduced this method in an earlier MR, @paulgessinger and I had a discussion about that as well. This method removes non-compatible tracks and returns a bool indicating if it was successful or not. I could rename the method in this PR, if you have a good suggestion.

Reply:
how about "removeIncompatibleTrack"?

Reply:
> how about "removeIncompatibleTrack"?
I'm not sure if I fully understand this name

Reply:
you could also call it removeTrackIfIncompatible

Reply:
I'll take that! Thanks!

Reply:
Done.